c10/core/SymFloat.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <c10/core/SymFloatNodeImpl.h>
+#include <c10/macros/Macros.h>
+#include <c10/util/Exception.h>
+#include <c10/util/intrusive_ptr.h>
+
+#include <limits>
+#include <memory>
+
+namespace c10 {
+
+// NB: this is actually double precision; we're using the Python naming here
+class C10_API SymFloat {
+ public:
+  /*implicit*/ SymFloat(double d) : data_(d){};
+  SymFloat(SymFloatNode ptr)
+      : data_(std::numeric_limits<double>::quiet_NaN()), ptr_(std::move(ptr)){};
```

Comment:
hm, so this deviates from SymInt with its negative number trick out of necessity due to bit properties of the types, and the net effect for now is the type is twice as big?

Reply:
I could NaN box it but there is no reason to right now, this is not perf critical.